x-pack/plugins/transform/public/app/sections/create_transform/components/source_index_preview/source_index_preview.tsx

```diff
@@ -106,6 +109,29 @@ export const SourceIndexPreview: React.FC<Props> = React.memo(({ indexPattern, q
     return { id, schema };
   });
 
+  const onSort = useCallback(
+    (sc: Array<{ id: string; direction: 'asc' | 'desc' }>) => {
+      const valid = sc.reduce((v, c) => {
+        if (v === false) return v;
+        const columnType = dataGridColumns.find(dgc => dgc.id === c.id);
+        const validSortingType = columnType?.schema !== 'json';
+        if (validSortingType === false) {
+          toastNotifications.addDanger(
+            i18n.translate('xpack.transform.sourceIndexPreview.invalidSortingColumnError', {
+              defaultMessage: `The column '{columnId}' cannot be used for sorting.`,
+              values: { columnId: c.id },
+            })
+          );
+        }
+        return validSortingType;
+      }, true);
```

Comment:
I find usage of `reduce` a bit confusing here: 
- it has a side effect inside 
- if one of the soring is invalid it doesn't check the others

It might be better to use `find` to get the first invalid one and show the toastNotifications after

Reply:
Tweaked in d17b83b.